plugin/parser/engine/pg/convert.go

```diff
@@ -196,11 +198,194 @@ func convert(node *pgquery.Node, text string) (res ast.Node, err error) {
 			}
 			return dropTable, nil
 		}
+	case *pgquery.Node_SelectStmt:
+		return convertSelectStmt(in.SelectStmt)
 	}
 
 	return nil, nil
 }
 
+func convertExpressionNode(node *pgquery.Node) (ast.ExpressionNode, []*ast.PatternLikeDef, []*ast.SubqueryDef, error) {
+	if node == nil || node.Node == nil {
+		return &ast.UnconvertedExpressionDef{}, nil, nil, nil
+	}
+	switch in := node.Node.(type) {
+	case *pgquery.Node_AConst:
+		return convertExpressionNode(in.AConst.Val)
+	case *pgquery.Node_String_:
+		return &ast.StringDef{Value: in.String_.Str}, nil, nil, nil
+	case *pgquery.Node_ResTarget:
+		return convertExpressionNode(in.ResTarget.Val)
+	case *pgquery.Node_TypeCast:
+		_, likeList, subqueryList, err := convertExpressionNode(in.TypeCast.Arg)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		return &ast.UnconvertedExpressionDef{}, likeList, subqueryList, nil
+	case *pgquery.Node_ColumnRef:
+		columnName := &ast.ColumnNameDef{Table: &ast.TableDef{}}
+		list := in.ColumnRef.Fields
+		switch len(in.ColumnRef.Fields) {
+		case 3:
```

Comment:
nit: define (length) constants for magic numbers 3, 2, 1. The name of the constants can be used as comments for better readability.

Reply:
The constant name is too short to explain the cases, so I add the detailed comments here instead.